web-console/src/components/auto-form.tsx

```diff
@@ -100,6 +103,15 @@ export class AutoForm<T> extends React.Component<AutoFormProps<T>, AutoFormState
     </HTMLSelect>
   }
 
+  private renderJSONInput(field: Field): JSX.Element {
+    const { model, onChange, JSONErrors, updateJSONErrors } = this.props;
+    return <JSONInput
+      value={(model as any)[field.name]}
+      onChange={(e: any) => onChange(Object.assign({}, model, { [field.name]: parseStringToJSON(e)}))}
+      updateErrors={(e: any) => updateJSONErrors!(Object.assign({}, JSONErrors, { [field.name]: validJson(e) || e === ''}))}
```

Comment:
why do you force TypeScript here with a `!` ? this is in fact an error and will throw an NPE if updateJSONErrors is not provided (it is optional in the API